enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk. These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+- Adds a top-level PlatformOperator cluster-scoped CRD.
+- Adds the Bundle and BundleDeployment OLM 1.x API(s).
+- Adds a validating webhook for the Bundle and BundleDeployment APIs.
+- Adds a CRD validation webhook.
+
+### Risks and Mitigations
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions that may require upfront RBAC to limit this API to admins.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will deliver through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Does the console need to be able to provide a UI experience for POs?
+- Does having separate APIs for ClusterOperators and PlatformOperators provide a degraded admin UX?
+- How to handle the maxOpenShiftVersion constraint given it's defined in the CSV resource?
+- Dedicated CatalogSource? Dedicated channel? The top-level PO API contains a catalog configuration?
+- Is it possible to have a dedicated PO index image?
```

Comment:
probably not, as we'd want vendors/partners/customers to be able to produce catalog images that contain POs

Reply:
What's the user story behind that? Are they adding their own new operators as POs? Or composing a collection of POs as a subset of what we provide? Or something else?

Reply:
> What's the user story behind that? Are they adding their own new operators as POs? 

That is the idea, at least at some point in the future.  E.g. a vendor who wants to provide their own ingress solution as a PO so they can provide a more holistic cluster lifecycle experience than what you could get if you just shipped it as an olm operator.

Reply:
i think we should move this to a resolved questions section w/ the answer being "no"

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/e6f356ef53203a97ab98f654704310f1e8a2f790.